### PR TITLE
Fix conversion of slices of binned data

### LINKED ISSUE #102: `convert` broken for slices of binned data
```python
scn.convert(new_data.copy(), 'tof', 'dspacing', scatter=True)
```
raises
```python
---------------------------------------------------------------------------
DimensionError                            Traceback (most recent call last)
<ipython-input-12-902261b1e9d2> in <module>
----> 1 scn.convert(sample['spectrum',0], 'tof', 'dspacing', scatter=True)

DimensionError: Cannot add coord exceeding DataArray dims
```

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #102 by making a copy if input is a slice.

### CODE REVIEW

File: neutron/convert.cpp [hunk 1]
```diff
@@ -3,22 +3,22 @@
 /// @file
 /// @author Simon Heybrock
 #include <set>
+#include <sstream>
```
Comment (OwenArnold) [inline neutron/convert.cpp:6]: What is this needed for?

**Review by OwenArnold — APPROVED** [review]

File: neutron/convert.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply neutron/convert.cpp:6]: You added it. I just moved it from below.

Comment (SimonHeybrock) [reply neutron/convert.cpp:6]: https://github.com/scipp/scippneutron/pull/73/files#r627401502

Comment (OwenArnold) [reply neutron/convert.cpp:6]: Ah sorry I didn't see the deletion and it didn't correspond as an include I could see was needed for any of the new changes.

**Pull request merged**